src/antares/service/local_services/binding_constraint_local.py

```diff
@@ -103,20 +95,76 @@ def _check_if_empty_ts(time_step: BindingConstraintFrequency, time_series: Optio
         time_series_length = (365 * 24 + 24) if time_step == BindingConstraintFrequency.HOURLY else 366
         return time_series if time_series is not None else pd.DataFrame(np.zeros([time_series_length, 1]))
 
-    def _write_binding_constraint_ini(self) -> None:
-        binding_constraints_ini_content = {
-            idx: idx_constraint.local_properties.list_ini_fields
-            for idx, idx_constraint in enumerate(self.binding_constraints.values())
-        }
-        self.ini_file.ini_dict = binding_constraints_ini_content
+    def _write_binding_constraint_ini(
+        self,
+        properties: BindingConstraintProperties,
+        constraint_name: str,
+        constraint_id: str,
+        terms: Optional[list[ConstraintTerm]] = None,
+    ) -> None:
+        """
+        Write a single binding constraint to the INI file, reconstructing a full BindingConstraintPropertiesLocal instance.
+
+        Args:
+            properties (BindingConstraintProperties): Basic properties of the binding constraint.
+            constraint_name (str): The name of the constraint.
+            constraint_id (str): The ID of the constraint.
+            terms (dict[str, ConstraintTerm], optional): Terms applying to the binding constraint. Defaults to None.
+        Raises:
+        BindingConstraintCreationError: If a binding constraint with the same name already exists in the INI file.
+        """
+
+        current_ini_content = self.ini_file.ini_dict or {}
+
+        if constraint_name in current_ini_content:
```

Comment:
It's a bit more complicated as we want to ensure we don't create 2 same **ids** as there will be a conflict inside the filesystem. Example: 2 BCs; bc_1 and BC_1 don't have the same name but have the same ids and therefore cannot co-exist on the fs so we have to raise an Exception in such case

To calculate the id without having to infer the way it's done we should probably introduce a static method inside a Constraint Object that will return the transform function. This way if we one day change the way a BC id is calculated the code will still work